lib/docker/artifact_image.js

```diff
@@ -70,32 +70,37 @@ export default class ArtifactImage {
 
     let originalTarball = path.join(downloadDir, 'image.tar');
 
-    let newTarball;
     try {
-      await downloadArtifact(
-        this.queue,
-        this.stream,
-        this.taskId,
-        this.artifactPath,
-        originalTarball,
-        this.runtime.dockerConfig
+      await this.runtime.stats.timeGen('taskImageDownloadTime',
```

Comment:
(really minor) I kind of feel like this is an abuse of Promises -- in a sync context, this would be `timeGen(() => downloadArtifact(..))`, making it clear that timeGen is responsible for the whole execution of downloadArtifact.  Here, timeGen is taking a Promise on the assumption that it was just begun, and timing it until resolution.

Totally not worth fixing -- I just find the interface weird.